### ISSUE TITLE
Incorrect type inference in combined enumerate, sorted, filter, with lambdas

### ISSUE THREAD

**Issue by simwr872** [microsoft/pyright#6042]
**Describe the bug**
Using `enumerate(sorted(filter(...` with custom lambdas for `filter` and `sorted` in the _**same**_ statement results in incorrect type inference.

**Code or Screenshots**
_Minimal example_
```python
from dataclasses import dataclass

@dataclass
class Item:
    age: int

items = [Item(15), Item(5), Item(10), Item(20)]

# Cannot access member "age" for type "object*"
#   Member "age" is unknown (reportGeneralTypeIssues)
enumerate(sorted(filter(lambda x: x.age > 7, items), key=lambda x: x.age))
```
Types are correct if the statement is broken up or if one of `sorted`/`filter` does not have a custom lambda function, see code below for some examples.
```python
from dataclasses import dataclass

@dataclass
class Item:
    age: int

items = [Item(15), Item(5), Item(10), Item(20)]

# Cannot access member "age" for type "object*"
#   Member "age" is unknown (reportGeneralTypeIssues)
enumeration = enumerate(sorted(filter(lambda x: x.age > 7, items), key=lambda x: x.age))
print(list(enumeration))

# multiple statements -> OK
filtered_items = filter(lambda x: x.age > 7, items)
enumeration = enumerate(sorted(filtered_items, key=lambda x: x.age))
print(list(enumeration))

sorted_items = sorted(filter(lambda x: x.age > 7, items), key=lambda x: x.age)
enumeration = enumerate(sorted_items)
print(list(enumeration))

filtered_items = filter(lambda x: x.age > 7, items)
sorted_items = sorted(filtered_items, key=lambda x: x.age)
enumeration = enumerate(sorted_items)
print(list(enumeration))

# simple types
items = [1,8,5,-2]

# enumeration is type `enumerate[Unknown]`
enumeration = enumerate(sorted(filter(lambda x: x > 0, items), key=lambda x: x))

# enumeration is type `enumerate[int]`
enumeration = enumerate(sorted(filter(lambda x: x > 0, items)))
# enumeration is type `enumerate[int]`
enumeration = enumerate(sorted(filter(None, items), key=lambda x: x))
# enumeration is type `enumerate[int]`
enumeration = enumerate(filter(lambda x: x > 0, items))
# enumeration is type `enumerate[int]`
enumeration = enumerate(sorted(items))
# sort is type `list[int]`
sort = sorted(filter(lambda x: x > 0, items), key=lambda x: x)
```

**VS Code extension or command-line**
Noticed error in VSCode, double checked with pyright as a command-line tool.
```sh
$ pyright --version
pyright 1.1.329
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.331, which I just published. It will also be included in a future release of pylance.
